db-server/test/backend/db_tests.js

```diff
@@ -2147,6 +2154,92 @@ module.exports = function (config, DB) {
       })
     })
 
+    describe('account recovery key', () => {
+      let account, data
+      beforeEach(() => {
+        account = createAccount()
+        account.emailVerified = true
+        return db.createAccount(account.uid, account)
+          .then(() => {
+            data = createRecoveryData()
+            return db.createRecoveryKey(account.uid, data)
+          })
+          .then((res) => {
+            assert.ok(res, 'empty response')
+          })
+      })
+
+      it('should fail to create for unknown user', () => {
+        return db.createRecoveryKey('12312312312', data)
+          .then(assert.fail, (err) => {
+            assert.equal(err.errno, 116, 'not found')
+          })
+      })
+
+      it('should create account recovery key', () => {
```

Comment:
This seems to duplicate the logic in the `beforeEach`, meaning it's not testing anything new.  Should it instead be trying to fetch the data that was set up in `beforeEach`, to ensure it was recorded correctly?

Reply:
Although you have a test below that does that, so maybe this one is just redundant?

Reply:
It is redundant, removed!